service/src/main/java/bio/terra/user/app/controller/ProfileApiController.java

```diff
@@ -19,32 +21,44 @@ public class ProfileApiController implements ProfileApi {
   private final HttpServletRequest servletRequest;
   private final ProfileService profileService;
   private final ProfileSamUserFactory userFactory;
+  private final SamService samService;
 
   @Autowired
   public ProfileApiController(
       HttpServletRequest servletRequest,
       ProfileService profileService,
-      ProfileSamUserFactory userFactory) {
+      ProfileSamUserFactory userFactory,
+      SamService samService) {
     this.servletRequest = servletRequest;
     this.profileService = profileService;
     this.userFactory = userFactory;
+    this.samService = samService;
   }
 
   @Override
-  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path) {
-    profileService.setProperty(getUser(), parsePath(path), body.getValue());
+  public ResponseEntity<AnyObject> setUserProfile(AnyObject body, String path, String userEmail) {
+    profileService.setProperty(targetIdOrSelf(userEmail), parsePath(path), body.getValue());
```

Comment:
targetIdOrSelf -> getProfileUser ?

Also I think it's a tad more readable to not inline:

```
String profileUser = getProfileUser(userEmail);
profileService.setProperty(profileUser, parsePath(path), body.getValue());
```